### DISCUSSION TITLE
Build with net8

### DISCUSSION THREAD

**Discussion by albenarammou** [community/community#88530] [Actions] [answered]
### Select Topic Area

Question

### Body

We start upgrade a net solution from 3.1.x to net 8.0.x
There are 5 project in the solution. 
When I work local the build is successful and I can run the net 8 solution.
But when running a yml file on GitHub the build starts with some of the projects, but the one which implements the EF 's DbContext and contains all data entities and migrations never ends and does not contains any other application code or business logic.
This project target  now is net 8.0 and was  netcoreapp3.1 
A part of the yml file

on:
  push:
    branches:
      - net8
  workflow_dispatch:

jobs:
  build:
    runs-on: windows-latest

    steps:
      - uses: actions/checkout@v3

      - name: Set up .NET Core
        uses: actions/setup-dotnet@v3
        with:
          dotnet-version: 8.0.x

      - name: Build with dotnet
        run: dotnet build x.sln --configuration Release

      - name: dotnet publish
        run: dotnet publish x.sln -c Release -o ${{env.DOTNET_ROOT}}/myapp

All restores are successful. Some of the projects are build and dll are created. The build of one project never end and we have the following message

The hosted runner: GitHub Actions 2 lost communication with the server. Anything in your workflow that terminates the runner process, starves it for CPU/Memory, or blocks its network access can cause this error.

Do you have any idea how can we find what the problem is and how can we solve it?

**Comment by albenarammou** [comment, marked as answer]
We solve it by implementing self-hosted runner

**Comment by ngocdinhmanh** [comment]
@albenarammou I have the same problem now, can you share more detail about the solution?

**Reply by albenarammou** [reply]
Hi, we realized that we need more power so we create a runner on windows-latest, 16-cores, 64GB RAM 600 GB SSD
https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/adding-self-hosted-runners

I hope this will help you.
